File: api-versioning/src/main/java/com/example/demo/PostRepository.java
package com.example.demo;

import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

import java.util.UUID;

public interface PostRepository {
    Mono<Post> findById(UUID id);
    Flux<Post> findAll();
    Mono<UUID> save(Post post);
    Mono<Void> update(UUID id, Post post);
    Mono<Void> deleteById(UUID id);
}

File: bean-registrar-kotlin-dsl/src/main/kotlin/com/example/demo/PostRepository.kt
package com.example.demo

import kotlinx.coroutines.flow.Flow
import java.util.*

interface PostRepository {
    suspend fun findById(id: UUID): Post?
    fun findAll(): Flow<Post>
    suspend fun save(post: Post): UUID
    suspend fun update(id: UUID, post: Post): Long
    suspend fun deleteById(id: UUID): Long
    suspend fun initializeSampleData(): Unit
}